### ISSUE TITLE
SDK should expose available WarningLevel values as MSBuild items

### ISSUE THREAD

**Issue by tmeschter** [dotnet/sdk#24570]
### Is your feature request related to a problem? Please describe.

The [.NET Project System](https://github.com/dotnet/project-system) tries to provide the user all the available options for the `WarningLevel` property, along with a concise description of each, like so:

<img width="510" alt="image" src="https://user-images.githubusercontent.com/10506730/160206941-711bb45d-f42a-4a60-b29a-f0bf802dfa16.png">

There are two problems with this:
1. The Project System will always be playing catch up with the SDK as newer versions of the SDK come out. If the `WarningLevel` is set to a value the Project System doesn't know about we'll often end up showing nothing at all because we don't know how to interpret the value.
2. If the user forces an older version of the SDK, the Project System may show them values that aren't meaningful. E.g., if the user forces the project to the .NET 5.0 SDK, we will still show "6" as an option because we don't know any better.

### Describe the solution you'd like

The SDK should provide the set of meaningful values, perhaps as a set of items:

``` XML
<ItemGroup>
  <AvailableWarningLevel Include="0" />
  <AvailableWarningLevel Include="1" />
  <AvailableWarningLevel Include="2" />
  <AvailableWarningLevel Include="3" />
  <AvailableWarningLevel Include="4" />
  <AvailableWarningLevel Include="5" />
  <AvailableWarningLevel Include="6" />
  <AvailableWarningLevel Include="7" />
  <AvailableWarningLevel Include="9999" />
</ItemGroup>
```

Or perhaps as a set of properties:

``` XML
<PropertyGroup>
  <AvailableWarningLevels>0;1;2;3;4;5;6;7;9999</AvailableWarningLevels>
</PropertyGroup>
```

The Project System could then populate the dropdown using the values, avoiding the need for a change to VS every time a new warning level becomes available.

It would also be great if the SDK could provide the localized description of each warning level, but if that is difficult to do the Project System could continue to handle that part. The worse that would happen is there would be times when we don't have a description for a particular warning level.

**Comment by baronfel** [comment]
This seems good, but WarningLevel is a language-agnostic concept and so both the list and the descriptions would need to be keyed by language identifiers in some way.

**Comment by Nirmal4G** [comment]
Can we put this into `Microsoft.Build.xsd` and `Microsoft.Build.CommonTypes.xsd` or a separate file just for SDKs inside the SDKs?

@rainersigwald thoughts?

**Comment by rainersigwald** [comment]
The project systems today don't have any way to pull from the `.xsd` files and do have the ability to pull from items, so I don't think that would be a great direction @Nirmal4G.

**Comment by Nirmal4G** [comment]
I kinda knew that! But my real question to you is that instead of making the build slow by putting UI data into MSBuild props/targets, putting them into `.xsd` should give you that perf advantage, right? Separation of concerns comes into play here, provided, as you said, that the project system uses the definition files instead.

We already know the advantage that JSON Schema provides for JSON, XSD, XSLT for XML, etc... Terminal uses JSON Schema to fill up the valid values and suggested values. Both Project Systems could have used the XSD from the start. Alas! things didn't play out that way but that does not mean it shouldn't now.

**Comment by Nirmal4G** [comment]
Yes, with conditional imports, CI builds will be unaffected. But what about local dev loop? That's important too. What I'm looking for, is to completely separate UI only things from build only things but still have that integrated feeling. Then everybody wins!
